grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/multiqueue/MultiQueueTaskManager.java

```diff
@@ -74,6 +77,7 @@ public MultiQueueTaskManager(EngineID engineId) {
 
         this.producer = ConfigHelper.kafkaProducer();
         EngineCacheProvider.init(EngineCacheDistributed.init(zookeeper));
+        LockProvider.init(new ZookeeperLock(zookeeper, LOCK));
```

Comment:
Since your lock provider can only house 1 lock can you allow for locking different tasks? For example updating counts needs a lock and post processing needs another lock. They should not be sharing locks.